### PR TITLE
Expanded alter table

### PR DESCRIPTION
**Pull request by madams51703**

### CODE REVIEW

File: plsql/PlSqlParser.g4 [hunk 1]
```diff
@@ -701,6 +701,18 @@ inline_constraint
       constraint_state?
     ;
 
+inline_ref_constraint
+    : SCOPE IS tableview_name
+      | WITH ROWID
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: Vertical bar `|` for the first-level alternatives should strictly follow after the colon `:`. Here and further.

```antlr
inline_ref_constraint
    : SCOPE IS tableview_name
    | WITH ROWID
```

File: plsql/PlSqlParser.g4 [hunk 2]
```diff
@@ -1052,16 +1114,188 @@ alter_table
...
+    : regular_id
+    ; 
+
+object_properties
+    : column_or_attribute (DEFAULT expression)? ( inline_constraint+ | inline_ref_constraint )
+    | ( out_of_line_constraint | out_of_line_ref_constraint | supplemental_logging_props )
+    ; 
+
+supplemental_logging_props
+    : SUPPLEMENTAL LOG (supplemental_log_grp_clause|supplemental_id_key_clause )
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: Space after and before the vertical bar.

File: plsql/PlSqlParser.g4 [hunk 3]
```diff
@@ -1052,16 +1114,188 @@ alter_table
...
+    ;
+
+partition_name
+    : regular_id
+    ; 
+
+object_properties
+    : column_or_attribute (DEFAULT expression)? ( inline_constraint+ | inline_ref_constraint )
+    | ( out_of_line_constraint | out_of_line_ref_constraint | supplemental_logging_props )
+    ;
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: Should be rewritten by the following way:

```antlr
object_properties
     : column_or_attribute (DEFAULT expression)? (inline_constraint+ | inline_ref_constraint)
     | out_of_line_constraint
     | out_of_line_ref_constraint
     | supplemental_logging_props
     ;
```

File: plsql/PlSqlParser.g4 [hunk 4]
```diff
@@ -1052,16 +1114,188 @@ alter_table
...
+
+nested_table_col_properties
+    : NESTED TABLE  (nested_item|COLUMN_VALUE) substitutable_column_clause? (LOCAL|GLOBAL)?
+       STORE AS tableview_name ( '(' ( '(' object_properties ')'
+                                     | physical_properties
+                                     | column_properties
+                                     )+
+                                  ')'
+                               )?
+        (RETURN AS? (LOCATOR|VALUE) )?
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: Space after and before the vertical bar.

File: plsql/PlSqlParser.g4 [hunk 5]
```diff
@@ -1052,16 +1114,188 @@ alter_table
...
+    : column_name datatype? (GENERATED ALWAYS)? AS '(' expression ')'
+        VIRTUAL? inline_constraint*
+    ;
+
+out_of_line_part_storage
+    : PARTITION partition_name 
+    ;
+
+nested_table_col_properties
+    : NESTED TABLE  (nested_item|COLUMN_VALUE) substitutable_column_clause? (LOCAL|GLOBAL)?
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: Space after and before the vertical bar.

File: plsql/PlSqlParser.g4 [hunk 6]
```diff
@@ -1052,16 +1114,188 @@ alter_table
...
+    : NOCOMPRESS
+    | COMPRESS (HIGH | MEDIUM | LOW )?
+    ;
+
+lob_retention_clause
+    : RETENTION (MAX | MIN UNSIGNED_INTEGER | AUTO | NONE )?
+    ;
+
+encryption_spec
+    : ( USING  CHAR_STRING )? (IDENTIFIED BY REGULAR_ID)? CHAR_STRING? ( NO? SALT )?
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: No space after and before the colon: `(USING  CHAR_STRING)?`

File: plsql/PlSqlParser.g4 [hunk 7]
```diff
@@ -827,8 +839,7 @@ build_clause
 
 parallel_clause
     : NOPARALLEL
-    | PARALLEL parallel_count=UNSIGNED_INTEGER
-      
+    | PARALLEL (parallel_count=UNSIGNED_INTEGER)?
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: Parenthesis can be omitted.

File: plsql/PlSqlParser.g4 (hunk 1 above)
Comment (madams51703) [reply plsql/PlSqlParser.g4]: that feedback is particularly appreciated, wasn't sure how to handle that situation

**Pull request merged**